### PR TITLE
155969884 retrieve publiccode file

### PR DESCRIPTION
**Pull request by r3vit**
Check and retrieve raw file (publiccode.yml) from bitbucket repositories.

### CODE REVIEW

File: cmd/all.go [hunk 1]
```diff
@@ -74,37 +79,66 @@ func startMetricsServer() {
 }
 
 func processRepositories(repositories chan crawler.Repository, processedCounter prometheus.Counter) {
-	ch := make(chan string, 100)
-
-	//throttle requests
-	rate := time.Second // 1 per second. TODO check rate limit (https://confluence.atlassian.com/bitbucket/rate-limits-668173227.html)
+	channelCapacity := 100
+	ch := make(chan string, channelCapacity)
+	counter := 0
```
Comment (lussoluca) [inline cmd/all.go]: counter is just used to fmt.Println on line 97 (which is commented), can we remove it?

File: cmd/all.go [hunk 2]
```diff
@@ -74,37 +79,66 @@ func startMetricsServer() {
...
 		go checkAvailability(repository.Name, repository.URL, ch, processedCounter)
+
+		// Comment: fmt.Println(counter)
+		counter = counter + 1
 	}
 }
 
 func checkAvailability(name, url string, ch chan<- string, processedCounter prometheus.Counter) {
+	// Retrieve the url.
 	response, err := http.Get(url)
```
Comment (lussoluca) [inline cmd/all.go]: can we use our httpclient to get those url also?

File: cmd/all.go [hunk 3]
```diff
@@ -74,37 +79,66 @@ func startMetricsServer() {
...
+	path := filepath.Join("./data", vendor, repo)
+
+	// MkdirAll will create all the folder path, if not exists.
+	if _, err := os.Stat(path); os.IsNotExist(err) {
+		os.MkdirAll(path, os.ModePerm)
+	}
+
+	err := ioutil.WriteFile(path+"/"+fileName, data, 0644)
+	if err != nil {
+		log.Info(err)
```
Comment (lussoluca) [inline cmd/all.go]: if this is an error it should be log.Error

File: cmd/all.go (hunk 1 above)
Comment (r3vit) [reply cmd/all.go]: Of course. Removed.

File: httpclient/httpclient.go [hunk 4]
```diff
@@ -21,33 +27,33 @@ func GetURL(URL string) ([]byte, error) {
...
 		if err != nil {
-			return nil, err
+			return nil, RespStatus{Status: err.Error(), StatusCode: -1}, err
 		}
 
 		// Check if the request results in http RateLimit error.
 		if resp.StatusCode == 429 {
 			sleep = sleep + (5 * time.Minute)
-			log.Infof("Rate limit reached, sleep %v minutes\n", sleep)
+			log.Error("Rate limit reached, sleep %v minutes\n", sleep)
```
Comment (lussoluca) [inline httpclient/httpclient.go]: in this case Info is correct, this is not an error

**Pull request merged**